iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -77,6 +77,14 @@ cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitia
         return cxx::error<ThreadError>(Thread::errnoToEnum(createResult.get_error().errnum));
     }
 
+    posixCall(iox_pthread_setname_np)(uninitializedThread->m_threadHandle, m_name.c_str())
+        .successReturnValue(0)
+        .evaluate()
+        .expect("This should never happen! Failed to set thread name.");
+    /// @todo thread specific comm file under /proc/self/task/[tid]/comm is read. Opening this file can fail
+    /// and errors possible for open(2) can be retrieved. Handle them here?
+    /// @todo Do we really want to terminate here?
```

Comment:
If not here, then where? I would say fail early, hence fail here.

Reply:
Maybe the comment is misleading. The question is if we want to terminate just because we cannot set (or below get) the thread name. It could be sufficient to log that the call failed but this depends somehow on how we handle potential errors of the underlying `open` call.

Reply:
I tend here to just inform the user that the thread-name could not be set. Normally, this does not restrict the functionality in any way.